apisix/plugins/hmac-auth.lua

```diff
@@ -343,6 +370,19 @@ local function get_params(ctx)
     params.date  = date or ""
     params.signed_headers = signed_headers and ngx_re.split(signed_headers, ";")
 
+    local keep_headers = get_keep_headers(params.access_key)
+    core.log.info("keep_headers: ", keep_headers)
+
+    if not keep_headers then
+        remove_headers(signature_key, algorithm_key, signed_headers_key)
+
+        if params.signed_headers then
```

Comment:
we do not need to remove the headers defined in `signed_headers`.
It may cause bugs.

Reply:
has changed.